modules/core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/validation/DocumentValidationException.java

```diff
@@ -61,11 +61,17 @@ public DocumentValidationReport getReport() {
     public String getMessage() {
         if (report.hasError()) {
             int num = report.numberOfErrors();
-            String violationMessage = report.asList().get(0).getMessage(null);
+            ValidationViolation violation = report.asList().get(0);
+            String xpath = "";
+            if (violation instanceof ConstraintViolation) {
+                ConstraintViolation cv = (ConstraintViolation) violation;
+                xpath = cv.getPathAsString();
```

Comment:
Hi, after this change I get an error message on a custom validation (in explorer code) which is not ideal (see https://github.com/nuxeo/nuxeo-explorer/blob/master/modules/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/repository/RepositoryDistributionSnapshot.java#L602).
The new error message is: `Constraint violation thrown on property : 'Distribution key or alias is reserved: 'current''`, see https://jenkins.platform.dev.nuxeo.com/job/nuxeo/job/nuxeo-explorer/job/master/237/testReport/junit/org.nuxeo.apidoc.test/TestSnapshotPersist/testUpdatePersisted/

This is a detail but can we use the old message when xpath is empty, for that kind of use cases? Note I could work around this by defining a constraint violation on my end to give a xpath here, but maybe we should also handle cases where the ValidationViolation is an instance of GlobalViolation, and not of ConstraintViolation, as expected here.

Reply:
Taken into account, created PR https://github.com/nuxeo/nuxeo/pull/4586 to fix this

Reply:
Thanks :+1: